src/Refactoring2-Transformations-Tests/RBInsertClassParametrizedTest.class.st

```diff
@@ -0,0 +1,133 @@
+	^ ParametrizedTestMatrix new
+		addCase: { #rbClass -> RBInsertClassRefactoring };
+		addCase: { #rbClass -> RBInsertClassTransformation };
+		yourself
+]
+
+{ #category : #accessing }
+RBInsertClassParametrizedTest >> constructor [ 
+	^ #addClass:superclass:subclasses:category:
+]
+
+{ #category : #running }
+RBInsertClassParametrizedTest >> setUp [
+
+	super setUp.
+	model := self abstractVariableTestData.
+]
+
+{ #category : #'failure tests' }
+RBInsertClassParametrizedTest >> testEmptyCategory [
+	self shouldFail: (self createRefactoringWithArguments:
+		{ #FooTest . #RBAbstractRefactoringTest . (Array with: self class) . #'' })
+]
+
+{ #category : #'failure tests' }
+RBInsertClassParametrizedTest >> testExistingClassName [
+	self shouldFail: (self createRefactoringWithArguments:
+		{ #Object . self class name asSymbol . #() .  #'Refactory-Testing' } )
+]
+
+{ #category : #tests }
+RBInsertClassParametrizedTest >> testInsertClassWithinExistingHiearchy [
+	| refactoring insertedClass parentOfInsertedClass childOfInsertedClass |
+	
+	refactoring := self createRefactoringWithArguments: 
+		{ #InsertedClass . #RBAbstractRefactoringTest . (Array with: self class) . #'Refactory-Testing' }.
+		"addClass: superclass: subclasses: category:"
+	self executeRefactoring: refactoring.
+	
+	insertedClass := refactoring model classNamed: #InsertedClass.
+	parentOfInsertedClass := refactoring model classNamed: #RBAbstractRefactoringTest.
+	childOfInsertedClass := refactoring model classNamed: self class name.
+	
+	"The inserted class is correctly inserted between the currrent class and its superclass.
+	The checks only check the inserted-superclass link."
+	self assert: insertedClass superclass equals: parentOfInsertedClass.
+	self assert: (parentOfInsertedClass subclasses includes: insertedClass).
+	
+	"Metaclasses are correctly linked"
+	self assert: insertedClass classSide superclass equals: parentOfInsertedClass classSide.
+	self assert: (parentOfInsertedClass classSide subclasses includes: insertedClass classSide).
+	
+	"The inserted class is correctly inserted between the currrent class and its superclass.
+	The checks only check the insertedclass and its subclasses."
+	self assert: childOfInsertedClass superclass equals: insertedClass.
+	self assert: (insertedClass subclasses includes: childOfInsertedClass).
+	self assert: childOfInsertedClass classSide superclass equals: insertedClass classSide.
+	self assert: (insertedClass classSide subclasses includes: childOfInsertedClass classSide)
+]
+
+{ #category : #'failure tests' }
+RBInsertClassParametrizedTest >> testModelExistingClassName [
+
+	self shouldFail: (self createRefactoringWithModel: model andArguments: 
+		{ #Foo . #Object . #() . #'Refactory-Testing' })
+]
+
+{ #category : #tests }
+RBInsertClassParametrizedTest >> testModelInsertClass [
+	| refactoring insertedClass superClass subclass |
+	superClass := model classNamed: #Foo.
+	subclass := model classNamed: #Bar.
```

Comment:
I changed these to `#Foo` and `#Bar` instead of `#Superclass` and `#Subclass` ([commit](https://github.com/pharo-project/pharo/pull/11757/commits/5d37a70ae71342c0fe587787476b892e054627cd)) because test data creates before mentioned classes. In order to use later ones we would need to change test data and update all tests that use them.